### DISCUSSION TITLE
Extend pre-trained model without load old trainedData

### DISCUSSION THREAD

**Discussion by mertdogan** [BrainJS/brain.js#720] [Q&A] [answered]
For checked example at https://scrimba.com/learn/neuralnetworks/neural-networks-example-simple-reinforcement-learning-cpkKyHd and it can be acceptable idea for small datasets to call them again and push new rows to trainingArray

but if you have very bir pre-trained model with huge dataset (and goes to bigger with new rows), it is not usable with them.

There must be a solution to add new data and train with it without call old trained datasets. 

Are there?

**Comment by repository23** [comment]
It seems that brain js do not have evolve function.
and keepnetwork intact seems buggy.

**Comment by robertleeplummerjr** [comment, marked as answer]
Call `.train` on a network that has been previously trained, or that has been instantiated from jSON will allow it to train again.  Note that it would be important to include both old training data and new, so long as it fits the criteria for previously trained data.

**Reply by ahmaddie1378** [reply]
ty for the amazing lib 
so if we call train() on a trained network it removes all old trains and retrains the network again?

**Comment by robertleeplummerjr** [comment]
No it does not so long as it was either trained or hydrated from
`net.fromJSON()`>

**Reply by ahmaddie1378** [reply]
ty so much for the answer
that is really nice 
i have another question 
what happens if we load from pre-trained json and dont call old data and just call train() with new data ?

**Reply by robertleeplummerjr** [reply]
It will alter the weights of the previously trained model to whatever the new data is.  This can be beneficial, but if the data doesn't include what the network was previously trained on, with the new training data, the network can and will eventually forget those training sets.

**Reply by ahmaddie1378** [reply]
now i understand ty for help 
so we put old data and new data into the same array and pass it to train() ?
or we use keepnetwork intact example (that one connects old data with the new one) ?
